### PR TITLE
Cg transfer microbial

### PR DESCRIPTION
**Pull request by barrystokman**

### CODE REVIEW

File: cg/meta/transfer/lims.py [hunk 1]
```diff
@@ -42,12 +49,23 @@ def __init__(self, status: Store, lims: LimsAPI):
...
+            MicrobialState.SEQUENCED: self.status.microbial_samples_to_sequence,
+            MicrobialState.DELIVERED: self.status.microbial_samples_to_deliver,
+        }
+
         self._date_functions = {
             SampleState.RECEIVED: self.lims.get_received_date,
             SampleState.PREPARED: self.lims.get_prepared_date,
             SampleState.DELIVERED: self.lims.get_delivery_date,
             PoolState.RECEIVED: self.lims.get_received_date,
-            PoolState.DELIVERED: self.lims.get_delivery_date,
```
Comment (ingkebil) [inline cg/meta/transfer/lims.py]: No more pools delivered? :)

Comment (ingkebil) [inline cg/meta/transfer/lims.py]: No more pools delivered? :)

File: cg/meta/transfer/lims.py [hunk 2]
```diff
@@ -114,6 +132,38 @@ def transfer_pools(self, status_type: PoolState):
...
+
+            lims_date = self._date_functions[status_type](microbial_sample_obj.internal_id)
+            statusdb_date = getattr(microbial_sample_obj, f'{status_type.value}_at')
+            if lims_date:
+
+                if statusdb_date and statusdb_date.date() == lims_date:
+                    continue
+
+                LOG.info(f"Found new {status_type.value} date for {microbial_sample_obj.internal_id}: " \
+                         f"{lims_date}, old value: {statusdb_date} ")
```
Comment (ingkebil) [inline cg/meta/transfer/lims.py]: This doesn't need to be `statusdb_date.date()`?

File: cg/meta/transfer/lims.py [hunk 3]
```diff
@@ -114,6 +132,38 @@ def transfer_pools(self, status_type: PoolState):
...
+
+                setattr(microbial_sample_obj, f"{status_type.value}_at", lims_date)
+                result.append((internal_id, lims_date))
```
Comment (ingkebil) [inline cg/meta/transfer/lims.py]: Seems unused?

File: cg/meta/transfer/lims.py [hunk 4]
```diff
@@ -114,6 +132,38 @@ def transfer_pools(self, status_type: PoolState):
                     else:
                         continue
 
+    def transfer_microbial_samples(self, status_type: MicrobialState):
+        """Transfer information about microbial samples."""
+
+        microbial_samples = self._microbial_samples_functions[status_type]()
+        result = []
```
Comment (ingkebil) [inline cg/meta/transfer/lims.py]: Seems unused?

File: cg/meta/transfer/lims.py (hunk 1 above)
Comment (barrystokman) [reply cg/meta/transfer/lims.py]: Fixed

File: cg/meta/transfer/lims.py (hunk 3 above)
Comment (barrystokman) [reply cg/meta/transfer/lims.py]: Leftover from debugging, removed

File: cg/meta/transfer/lims.py (hunk 4 above)
Comment (barrystokman) [reply cg/meta/transfer/lims.py]: Leftover from debugging, removed

File: cg/meta/transfer/lims.py (hunk 1 above)
Comment (barrystokman) [reply cg/meta/transfer/lims.py]: Fixed

**Comment by ingkebil** [comment]
Looking good! We have enabled coveralls which requires 50% coverage. This PR puts us dangerously close to that number :hole:

File: cg/meta/transfer/lims.py (hunk 2 above)
Comment (barrystokman) [reply cg/meta/transfer/lims.py]: statusdb_date is NoneType if no date is found and therefore has no date() method, that's probably why it's used like that in transfer_samples where I blatantly copied this from :)

**Comment by ingkebil** [comment]
I hope you don't mind that I deploy this one already

**Pull request merged**